app/controllers/api/v1/users_controller.rb

```diff
@@ -114,7 +117,9 @@ def change_password
       private
 
       def user_params
-        params.require(:user).permit(:name, :email, :password, :avatar, :language, :role_id, :invite_token)
+        return @user_params unless @user_params.nil?
```

Comment:
With our current way of using the `ActionController::Parameters` API we're having two issues:
1. The filtering operation is required only once, for each subsequent call that is a waste of computation and not required delay.
2. Each call creates a new instance.
3. Because of 2. the params methods will give an immutable instance.
4. Because of 3. errors can be created easily without being caught (e.g. expecting that calling user_params[:language] = I18n.default_locale will change the user language which will not do).

Reply:
If you're okay with this, then I believe we may want to generalize this throughout the codebase.

Reply:
Can we not jsut do `@user_params ||= params.require...`

Reply:
Is this common? I don't think I've seen this

Reply:
@farhatahmad sure,

Reply:
@scouillard what exactly?

Reply:
@scouillard we may want to generalize this throughout the codebase to reduce error surface and to improve performance even when we're not mutating the params since there's always a great chance of reusing filtered params in the same request response cycle.